docs/src/examples/darcy.md

```diff
@@ -0,0 +1,138 @@
+const GRF = GaussianRandomFields
+```
+
+and finally the EKP packages.
+
+```julia
+using EnsembleKalmanProcesses
+using EnsembleKalmanProcesses.ParameterDistributions
+const EKP = EnsembleKalmanProcesses
+```
+
+We include the forward solver here.
+
+```julia
+include("GModel.jl")
+```
+
+We define the spatial domain and discretization,
+
+```julia
+rng = Random.MersenneTwister(seed)
+dim = 2
+N, L = 80, 1.0
+pts_per_dim = LinRange(0, L, N)
+```
+
+To provide a simple test case, we assume that the true function parameter is a particular sample from the function space we set up to define our prior. We choose a value of the truth that doesnt have a vanishingly small probability under the prior defined by a probability distribution over functions; taken to be a family of Gaussian Random Fields (GRF). This function distribution is characterized by a covariance function (Matern) and an appropriate representation (Karhunen-Loeve expansion). The representation is truncated to a finite number of coefficients, the degrees of freedom (`dofs`), which define the effective dimension of the learning problem that is decoupled from the spatial discretization. Larger `dofs` may be required to represent multiscale functions, but come at an increased dimension of the parameter space and therefore a typical increase in cost and difficulty of the learning problem.
+
+```julia
+smoothness = 2.0
+corr_length = 0.5
+dofs = 50
+
+grf = GRF.GaussianRandomField(
+    GRF.CovarianceFunction(dim, GRF.Matern(smoothness, corr_length)),
+    GRF.KarhunenLoeve(dofs),
+    pts_per_dim,
+    pts_per_dim,
+)
+```
+
+We define a wrapper around the GRF, and as the permeability field must be positive we introduce a domain constraint into the function distribution. 
+
+```julia
+pkg = GRFJL()
+distribution = GaussianRandomFieldInterface(grf, pkg) # our wrapper from EKP
+domain_constraint = bounded_below(0) # make κ positive
+pd = ParameterDistribution(
+    Dict("distribution" => distribution, "name" => "kappa", "constraint" => domain_constraint),
+) # the fully constrained parameter distribution
+```
+
+Henceforth, the GRF is interfaced in the same manner as any other parameter distribution with regards to interface. We sample a reasonably high-probability value from this distribution as a true value (here all degrees of freedom set with ``u_{\mathrm{true}} = -1.5``). We then use the EKP transform function to build the corresponding instance of the ``\kappa_{\mathrm{true}}``.
+
+```julia
+u_true = -1.5 * ones(dofs,1) # the truth parameter
+κ_true = transform_unconstrained_to_constrained(pd, u_true) # builds and constrains the function.
+κ_true = reshape(κ_true, N, N)
+```
+
+We generate the data sample for the truth in a perfect model setting by evaluating the the model here, and observing the flow velocity at a few subsampled points in each dimension (here `obs_ΔN` leads to a ``7 \times 7`` observation grid), and we assume 5% additive observational noise on the measurements.
```

Comment:
I would explain the observation grid a little more, since going from obs_ΔN = 10 to a 7x7 observation grid is not intuitive.